From 49d3cd0f299dc73185329e94ef5052ae731b8071 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@st.com>
Date: Fri, 7 Feb 2020 11:30:53 +0100
Subject: [PATCH 04/88] rtc: rtc-stm32: change PM callbacks to "_noirq()"

The rtc drivers stop the RTCAPB clock during suspend, but the
irq handle from RTC is called before starting clock. Then we are
blocked during IP access.
As the driver is registered as wakeup_source we can change PM
callbacks to '_nor_irq()' to resume automatically.

Upstream-Status: Denied Downstream stm32-gh v5.4-stm32mp [11f5efd00392116bfc30c0288c6dffd13144c44b]
Change-Id: I3cd8d0e145e5ff2b1c19d9d6d8fe1d3f7f7188a1
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/157299
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: CIBUILD <smet-aci-builds@lists.codex.cro.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@st.com>
---
 drivers/rtc/rtc-stm32.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/rtc/rtc-stm32.c b/drivers/rtc/rtc-stm32.c
index 6b01986745bcd..3bfddc691dd3d 100644
--- a/drivers/rtc/rtc-stm32.c
+++ b/drivers/rtc/rtc-stm32.c
@@ -897,8 +897,9 @@ static int stm32_rtc_resume(struct device *dev)
 }
 #endif
 
-static SIMPLE_DEV_PM_OPS(stm32_rtc_pm_ops,
-			 stm32_rtc_suspend, stm32_rtc_resume);
+static const struct dev_pm_ops stm32_rtc_pm_ops = {
+	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_rtc_suspend, stm32_rtc_resume)
+};
 
 static struct platform_driver stm32_rtc_driver = {
 	.probe		= stm32_rtc_probe,
-- 
2.42.0

